File: 프로그래머스/lv3/12987. 숫자 게임/숫자 게임.java
import java.util.Arrays;

class Solution {
    public int solution(int[] A, int[] B) {
        Arrays.sort(A);
        Arrays.sort(B);

        int ia = 0;
        int ib = 0;
        int answer = 0;


        while (ia < A.length && ib < B.length) {
            if (A[ia] < B[ib]) {
                ia++;
                answer++;
            }
            ib++;
        }

        return answer;
    }
}

File: 프로그래머스/lv3/12987. 숫자 게임/숫자 게임.js
function solution(A, B) {
    A.sort((a, b) => b - a);
    B.sort((a, b) => b - a);

    let j = 0; // B를 가리키는 인덱스
    let ans = 0; // 점수

    for (let i = 0; i < A.length; i++) { // i는 A를 가리키는 인덱스
        if (A[i] < B[j]) { // B가 더 클 때 
            ans++; // 점수 증가
            j++; // B를 가리키는 인덱스 증가
        }
    }

    return ans;
}
